pep-0458.txt

```diff
@@ -527,243 +789,247 @@ How Should Metadata be Generated?
-distributions.  That is, there needs to be a way to ensure consistency of
-metadata and repository files when multiple developers simultaneously change the
-same metadata or distributions.  There are also issues with consistency on PyPI
+target files.  That is, there needs to be a way to ensure consistency of
+metadata and target files when multiple developers simultaneously change these
+files. There are also issues with consistency on PyPI
 without TUF, but the problem is more severe with signed metadata that MUST keep
 track of the files available on PyPI in real-time.
 
-Suppose that PyPI generates a *snapshot*, which indicates the latest version of
-every metadata except *timestamp*, at version 1 and a client requests this
+Suppose that PyPI generates a *snapshot* that indicates the latest version of
+every metadata, except *timestamp*, at version 1 and a client requests this
 *snapshot* from PyPI.  While the client is busy downloading this *snapshot*,
 PyPI then timestamps a new snapshot at, say, version 2.  Without ensuring
 consistency of metadata, the client would find itself with a copy of *snapshot*
-that disagrees with what is available on PyPI, which is indistinguishable from
-arbitrary metadata injected by an attacker.  The problem would also occur for
+that disagrees with what is available on PyPI. The result would be indistinguishable from
+arbitrary metadata injected by an attacker.  The problem would also occur with
 mirrors attempting to sync with PyPI.
 
-
 Consistent Snapshots
 --------------------
 
-There are problems with consistency on PyPI with or without TUF.  TUF requires
-that its metadata be consistent with the repository files, but how would the
-metadata be kept consistent with projects that change all the time?  As a
-result, this proposal MUST address the problem of producing a consistent
-snapshot that captures the state of all known projects at a given time.  Each
-snapshot should safely coexist with any other snapshot, and be able to be
-deleted independently, without affecting any other snapshot.
+To keep TUF metadata on PyPI consistent with the highly volatile target files,
+consistent snapshots SHOULD be used. Each consistent snapshot captures the
+state of all known projects at a given time and MAY safely coexist with any
+other snapshot, or be deleted independently, without affecting any other
+snapshot.
 
-The solution presented in this PEP is that every metadata or data file managed
-by PyPI and written to disk MUST include in its filename the `cryptographic
-hash`__ of the file.  How would this help clients that use the TUF protocol to
-securely and consistently install or update a project from PyPI?
+To maintain consistent snapshots, all TUF metadata MUST, when written to disk,
+include a version number in their filename:
 
-__ https://en.wikipedia.org/wiki/Cryptographic_hash_function
+  VERSION_NUMBER.ROLENAME.json,
+    where VERSION_NUMBER is an incrementing integer, and ROLENAME is one of the
```

Comment:
Does this have to be an incrementing integer? My concern here is simply that an ever increasing integer can run into scalability problems, particularly if people don't take into account it's ever increasing nature (example, needing a different integer size in a database).

My second question is, do we have to guarantee these are *never* reused, or can they be reused after some time has passed? 

My gut tells me we'd be better off using a uuid for a version number, or generating a random string of characters. The biggest downside I see to that is the extremely remote chance of generating the same version number twice-- but that can be solved by recording all possible versions ever used (which might be good to do anyways to return a 410 instead oof a 404).

Reply:
Addressed by recommending a big integer and referencing how TUF protects against fast-forward attacks in https://github.com/secure-systems-lab/peps/pull/72.